### DISCUSSION TITLE
is there a script to find the center line of a scanned text image

### DISCUSSION THREAD

**Discussion by shenenw** [ImageMagick/ImageMagick#4937] [Help]
Hi,
This is the first time I post a question, I also Googled it and I don't find a solution or anybody asking a similar questions.
When you scan a book , it's usually 2 pages per scan. In my case it's landscape orientation with one page on top and the other below it.

First I extract bottom half by specify crop window(2760x1680) followed by a starting point coordinate(15,1670):
for i in {2..200..2}  
do
#extract bottom half of the page:
echo "convert p$i-$[i+1].tif -crop 2760x1680+15+1670 ./converted-single/p$i.tif"

 #top half
echo "convert p$i-$[i+1].tif -crop 2760x1680+8+30 ./converted-single/p$[i+1].tif"
done

It all works fine.
However, when you scan it, you usually align one side of the book on the scanner(usually left).
As you flip the page, one side of the book getting thicker while the other side of the book getting thinner.
This will slightly change the scanned window location(harder to align).  So the same crop dimension on the script works for 100 pages or so will become slightly off.

The solution I can think of is to detect the center line coordinate of the scanned image(2 pages per scan).  Because the distance from the center line to left and right of the page will be the same all the time, I can then use that information to adjust the starting point of the crop window.
Or other ideas.
Thanks.

**Comment by fmw42** [comment]
Average the image down to one row. Then look for the darkest location near the middle that would be the centerline shadow.

**Comment by snibgo** [comment]
I suspect the input should be reduced to one column, not one row.

@shenenw: It would help if you showed a sample scan.

**Reply by fmw42** [reply]
I thought he said the page was going to be left right. So one row.  If top bottom scanning, then one column is correct.

**Comment by snibgo** [comment]
Could be. I find the OP confusing, so a sample scan would help.

**Reply by fmw42** [reply]
Agreed

**Reply by shenenw** [reply]
Thanks for the reply.  I prepared a sample, but I don't see a way to attach it(maybe there isn't, only external link allowed, my first post, still learning).   So I put it in Gxxgle and here is the [link](https://drive.google.com/file/d/14r_GBXMJ3hbWKgb07e4U-z8x3wRxJibx/view?usp=sharing)

**Reply by fmw42** [reply]
Just drag and drop into these windows to upload

**Reply by fmw42** [reply]
The scan is top bottom, so one would average down to one column and look for the dark center line.
